geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3BlobStoreConfigProvider.java

```diff
@@ -41,20 +48,53 @@ public Object fromString(String str) {
 
         @Override
         public Object fromString(String str) {
+            str = resolveFromEnv(str);
             if (Strings.isNullOrEmpty(str)) {
                 return null;
             }
             return super.fromString(str);
         }
     };
 
+    private static SingleValueConverter NullableStringConverter = new StringConverter() {
```

Comment:
This is not just a "nullable" converter, it's more of a "EnvironmentStringConverter", it's main job is to pull info from the environment. As coded it might also unwillingly pick values that were not meant to be variable, as it's "always on".
In GeoServer we have a flag to control parameter injection from the environment (ALLOW_ENV_PARAMETRIZATION) and GeoWebCache has the same here: [GeoWebCacheEnvironment](https://github.com/GeoWebCache/geowebcache/blob/926e5e8ded827d35b3cbfd353af36387cde50f29/geowebcache/core/src/main/java/org/geowebcache/GeoWebCacheEnvironment.java#L45)
Imho that should be used, following an approach similar to the existing ones for uniformity:
https://github.com/GeoWebCache/geowebcache/blob/73afa58a2154d2ef0e223812cf8b8569e1b9d51b/geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/ConfigLoader.java#L292

Reply:
Sorry, did not see the "resolveFromEnv" code, my apologies!
The name could still use some love though :-)